tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -11939,6 +11945,44 @@ SpirvInstruction *SpirvEmitter::processIntrinsicUsingGLSLInst(
   return nullptr;
 }
 
+SpirvInstruction *SpirvEmitter::processEvaluateAttributeAt(
+    const CallExpr *callExpr, hlsl::IntrinsicOp opcode, SourceLocation loc,
+    SourceRange range) {
```

Comment:
Should `loc` and `range` should be const refs? So like: `const SourceLocation& loc, const SourceRange& range`

Reply:
The standard idiom in LLVM is to pass them by value. SourceLocation is a class with just a single member, so that can be passed in a register. SourceRange is two SourceLocations, so I believe it can be passed in a register as well. However, that depends on the ABI.

Reply:
ACK.